File: Day05_Invertedhalfpyramid.py
"""
Day 05: Inverted Half Pyramid
Printing a triangle of stars upside down.
"""

n = int(input("Enter the value of n: "))

# Outer loop goes downwards from n to 1
# range(start, stop, step) -> range(n, 0, -1) stops before 0 (at 1)
for i in range(n, 0, -1):
    for j in range(1, i + 1):
        print("*", end="")
    print()

File: Day05_Invertedhalfpyramid.java

/**
 * Day 05: Inverted Half Pyramid
 * Printing a triangle of stars upside down.
 */

import java.util.*;

public class Day05_Invertedhalfpyramid {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the value of n");
        int n = sc.nextInt();
        for (int i = n; i >= 1; i--) {
            for (int j = 1; j <= i; j++) {
                System.out.print("*");
            }
            System.out.println();
        }
    }
}
